packages/core/src/boot/init.ts

```diff
@@ -51,6 +52,12 @@ export function commonInit(initConfiguration: InitConfiguration, buildEnv: Build
   const configuration = buildConfiguration(initConfiguration, buildEnv)
   const internalMonitoring = startInternalMonitoring(configuration)
 
+  const enableExperimentalFeatures = Array.isArray(initConfiguration.enableExperimentalFeatures)
+    ? initConfiguration.enableExperimentalFeatures
+    : []
+
+  setEnabledExperimentalFeatures(enableExperimentalFeatures)
+
```

Comment:
I think it would be better to move this at the beginning of `commonInit`, so we can use feature flags while building the configuration.